src/options.cpp

```diff
@@ -1035,6 +1038,38 @@ std::vector<options_manager::id_and_option> options_manager::build_soundpacks_li
     return result;
 }
 
+std::vector<std::string> options_manager::get_lang_list()
+{
+    std::vector<std::string> lang_dirs = get_directories_with( PATH_INFO::lang_file(),
+                                         PATH_INFO::langdir(), true );
+    const std::string start_str = "mo/";
+    const std::size_t start_len = start_str.length();
+    const std::string end_str = "/LC_MESSAGES";
+    std::for_each( lang_dirs.begin(), lang_dirs.end(), [&]( std::string & dir ) {
+        const std::size_t start = dir.find( start_str ) + start_len;
+        const std::size_t len = dir.rfind( end_str ) - start;
+        dir = dir.substr( start, len );
+    } );
+    return lang_dirs;
+}
+
+std::vector<options_manager::id_and_option> options_manager::get_actual_lang_options()
+{
+    std::vector<std::string> lang_list = options_manager::get_lang_list();
```

Comment:
Better to make `lang_list` a `std::unordered_set` of `std::string`s. It will make your `copy_if` predicate simpler, and ensure that this function is O(n) in the number of languages, rather than O(n^2).